proto/tendermint/farming/v1beta1/proposal.proto

```diff
@@ -14,10 +15,112 @@ message PublicPlanProposal {
   option (gogoproto.goproto_getters)  = false;
   option (gogoproto.goproto_stringer) = false;
 
-  string title       = 1;
+  // title specifies the title of the plan
+  string title = 1;
+
+  // description specifies the description of the plan
   string description = 2;
 
-  // plans specifies the plan interface(s); it can be FixedAmountPlan or
-  // RatioPlan
-  repeated google.protobuf.Any plans = 3;
+  // name specifies the plan name that describes what plan is for
+  string name = 3;
+
+  // add_request_proposals specifies AddRequestProposal object
+  repeated AddRequestProposal add_request_proposals = 4 [(gogoproto.moretags) = "yaml:\"add_request_proposals\""];
+
+  // update_request_proposals specifies UpdateRequestProposal object
+  repeated UpdateRequestProposal update_request_proposals = 5
+      [(gogoproto.moretags) = "yaml:\"update_request_proposals\""];
+
+  // delete_request_proposals specifies DeleteRequestProposal object
+  repeated DeleteRequestProposal delete_request_proposals = 6
+      [(gogoproto.moretags) = "yaml:\"delete_request_proposals\""];
+}
+
+// AddRequestProposal details a proposal for creating a public plan.
+message AddRequestProposal {
+  // farming_pool_address defines the bech32-encoded address of the farming pool
+  string farming_pool_address = 1 [(gogoproto.moretags) = "yaml:\"farming_pool_address\""];
+
+  // termination_address defines the bech32-encoded address that terminates plan
+  // when the plan ends after the end time, the balance of farming pool address
+  // is transferred to the termination address
+  string termination_address = 2 [(gogoproto.moretags) = "yaml:\"termination_address\""];
+
+  // staking_coin_weights specifies coin weights for the plan
+  repeated cosmos.base.v1beta1.DecCoin staking_coin_weights = 3 [
+    (gogoproto.moretags)     = "yaml:\"staking_coin_weights\"",
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.DecCoins",
+    (gogoproto.nullable)     = false
+  ];
+
+  // start_time specifies the start time of the plan
+  google.protobuf.Timestamp start_time = 4
+      [(gogoproto.stdtime) = true, (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"start_time\""];
+
+  // end_time specifies the end time of the plan
+  google.protobuf.Timestamp end_time = 5
+      [(gogoproto.stdtime) = true, (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"end_time\""];
+
+  // epoch_amount specifies the distributing amount for each epoch
+  repeated cosmos.base.v1beta1.Coin epoch_amount = 6 [
+    (gogoproto.moretags)     = "yaml:\"epoch_amount\"",
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (gogoproto.nullable)     = false
+  ];
+
+  // epoch_ratio specifies the distributing amount by ratio
+  string epoch_ratio = 7 [
+    (gogoproto.moretags)   = "yaml:\"epoch_ratio\"",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable)   = false
+  ];
+}
```

Comment:
It would be nice to add a description that only one of these fields needs to be included and that each one is created as FixedAmountPlan and RatioPlan.

And Could you add these structs with descriptions and an example to `spec/09_proposals.md`?